cli/internal/gcp/client/instances.go

```diff
@@ -172,10 +179,12 @@ func (c *Client) TerminateInstances(ctx context.Context) error {
 		c.controlPlaneInstanceGroup = ""
 		c.controlPlanes = make(cloudtypes.Instances)
 	}
-	if err := c.waitForOperations(ctx, ops); err != nil {
-		return err
-	}
-	ops = []Operation{}
+
+	return c.waitForOperations(ctx, ops)
```

Comment:
Won't this refactoring slow down the termination because it unnecessarily blocks here?

Reply:
It waited before, still waits and it has to wait, because the templates can't be deleted as long as the instance group manager that uses it still exist.

Reply:
I see. I didn't get the diff correctly and thought that there was only one wait before for all ops that now would be two waits.